bls/src/main/java/tech/pegasys/teku/bls/BLSSecretKey.java

```diff
@@ -76,11 +76,6 @@ public Bytes32 toBytes() {
     return secretKey.toBytes();
   }
 
-  /** Overwrites the key with zeros so that it is no longer in memory */
-  public void destroy() {
-    secretKey.destroy();
-  }
```

Comment:
Not sure if anyone is using this method now, but may be it's worth leaving it? It's good to minimize the time a secret key is in memory. Also when using it a secret key is less likely to occur in a heap dump.

Reply:
Agreed. I used it [here](https://github.com/ConsenSys/Ecies/blob/master/src/main/java/tech/pegasys/ecies/Ecies.java#L43) for example. We should assume that others might use our library, and support basic hygiene where it's easy to do so.

Reply:
I think it got removed because there isn't a `destroy` method in BLST anymore.  I've restored it and just overwritten the key with zeros.

That said, SWIG does some funky stuff so that all the data lives in the JVM heap (in the `long[] swigCPtr` variable which is actually the data and not a pointer).  So I can't tell if a write will actually overwrite the content of the `long[]` or just replace it with a new array.  I think it overwrites which is what we need but can't be sure.

Reply:
> So I can't tell if a write will actually overwrite the content of the long[] or just replace it with a new array. I think it overwrites which is what we need but can't be sure.

Just checked, yes it overrides